src/v/s3/client.h

```diff
@@ -202,6 +203,31 @@ class client {
       const object_key& key,
       const ss::lowres_clock::duration& timeout);
 
+    struct delete_objects_result {
+        struct key_reason {
+            object_key key;
+            std::string reason;
```

Comment:
this should probably be an `ss::sstring`

Reply:
done